packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +371,123 @@ export class CloudFormationDeployments {
+    sdk: ISDK,
+  ): Promise<void> {
+    const listStackResources = parentStackName ? new LazyListStackResources(sdk, parentStackName) : undefined;
+    const parentStackResources = Object.entries(generatedParentTemplate.Resources ?? {});
+    for (const [nestedStackLogicalId, nestedStackResource] of parentStackResources) {
+      if (!this.isCdkManagedNestedStack(nestedStackResource)) {
+        continue;
+      }
+
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResources, sdk,
+      );
+
+      if (!generatedParentTemplate.Resources[nestedStackLogicalId].Properties) {
+        generatedParentTemplate.Resources[nestedStackLogicalId].Properties = {};
+      }
+      generatedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate = nestedStackTemplates.generatedNestedTemplate;
+
+      if (!deployedParentTemplate.Resources) {
+        deployedParentTemplate.Resources = {};
+      }
+      if (!deployedParentTemplate.Resources[nestedStackLogicalId]) {
+        deployedParentTemplate.Resources[nestedStackLogicalId] = {};
+        deployedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+      }
+      if (!deployedParentTemplate.Resources[nestedStackLogicalId].Properties) {
+        deployedParentTemplate.Resources[nestedStackLogicalId].Properties = {};
+      }
+      deployedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate = nestedStackTemplates.deployedNestedTemplate;
+
+      await this.replaceNestedStacksInParentTemplate(
+        rootStackArtifact,
+        generatedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate,
+        deployedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate ?? {},
+        nestedStackTemplates.nestedStackName,
+        sdk,
+      );
+    }
+  }
+
+  private async getNestedStackTemplates(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
+    parentStackName: string | undefined, listStackResources: ListStackResources | undefined, sdk: ISDK,
+  ): Promise<NestedStackTemplates> {
+    const nestedTemplatePath = path.join(rootStackArtifact.assembly.directory, nestedTemplateAssetPath);
+    const nestedStackArn = await this.getNestedStackArn(nestedStackLogicalId, parentStackName, listStackResources);
+
+    // CFN generates the nested stack name in the form `ParentStackName-NestedStackLogicalID-SomeHashWeCan'tCompute,
+    // the arn is of the form: arn:aws:cloudformation:region:123456789012:stack/NestedStackName/AnotherHashWeDon'tNeed
+    // so we get the ARN and manually extract the name.
+    const nestedStackName = nestedStackArn?.slice(nestedStackArn.indexOf('/') + 1, nestedStackArn.lastIndexOf('/'));
```

Comment:
Why did you group these

```
nestedTemplatePath
nestedStackArn

nestedStackName
```

? `nestedTemplatePath` and `nestedStackArn` have nothing in common, while `nestedStackArn` and `nestedStackName` have everything in common.

I think a much more logical grouping would be:

```
nestedTemplatePath

nestedStackArn
nestedStackName
```